api/src/main/java/org/apache/iceberg/io/FileIO.java

```diff
@@ -66,4 +66,10 @@ default void deleteFile(OutputFile file) {
    */
   default void initialize(Map<String, String> properties) {
   }
+
+  /**
+   * Close File IO and release underlying resources.
+   */
+  default void close() {
```

Comment:
Nit: Possibly add some documentation here about whether or not `close` needs to be called? Currently it's somewhat confusing (as the underlying issue is somewhat confusing), but end user's are exposed this API and encouraged / allowed to write their own FileIO per the docs, so I feel that the requirements should be clearly laid out.

Reply:
I agree w. @kbendick. I don't think we should extend `AutoCloseable` *and* add a default impl.

I think there is a good argument to have `FileIO` implement `Closeable` (rather than `Autocloseable` as its fairly explicitly related to IO), what do you think @rdblue? But I would make `FileIO` implementations implement `close` rather than add a default here, otherwise its not clear to implementers that they may have to deal w/ closing their resources.

Reply:
Thanks for the feedback! I will update the comment but for the default impl, I mentioned this a little bit in the other response I had, the reason for me to add a default implementation here was because I think this use case of having to call `close` is rare and for the majority of the use case it is not needed, and because of this, to me breaking compatibility of custom FileIO implementation by adding a non-default method in the interface might not be worth it. Please let me know if you have more feedback!